csrc/scheduler/resize.cpp

```diff
@@ -373,15 +428,55 @@ void ResizeScheduler::schedule(Fusion* fusion, const HeuristicParams* params) {
   // [..., I0/bdimx(BIDx), bdimx(TIDx), vec_factor]
 
   // Propagate the reference to the other tensors. Note that the
-  // update flag is enabled so to workaround the resize propagation
+  // update flag is enabled to workaround the resize propagation
   // issue. This may not work if there's a tensor that is reshaped
   // from the reference tensor, but that should not be the case as the
   // reference is picked by the same routine used for the pointwise
   // scheduler.
-  scheduler_tools::scheduleLoopDomainsLike(
-      fusion->allTvs(),
-      ref_tv->getLoopDomain(),
-      /*update_loop_domain_only=*/true);
+  //
+  // When an ending static repeat is detected and the repeat ID is
+  // moved to the outermost position, propagation is done separately
+  // between the tensors before the repeat and after the repeat. The
+  // tensors are first grouped into the pre-repeat group and the
+  // post-repeat group, where only the latter group has the repeat
+  // IDs. When propagating the loop domain of the reference tensor,
+  // which has the repeat ID, the full loop domain is propagated only
+  // to the post-repeat group. For the pre-repeat group, the repeat ID
+  // is dropped and only the remaining loop domain is propagated.
+  if (repeat_id_moved_to_outermost) {
+    // Divide all tvs to the pre and posgt repeat groups
+    auto all_tvs = fusion->allTvs();
+    std::vector<TensorView*> post_repeat_tvs;
+    post_repeat_tvs.reserve(static_repeat_info->repeat_tvs.size());
+    std::vector<TensorView*> pre_repeat_tvs;
+    pre_repeat_tvs.reserve(
+        all_tvs.size() - static_repeat_info->repeat_tvs.size());
+    for (auto tv : all_tvs) {
+      if (static_repeat_info->repeat_tvs.count(tv)) {
+        post_repeat_tvs.push_back(tv);
+      } else {
+        pre_repeat_tvs.push_back(tv);
+      }
+    }
+
+    // The repeat ID should be located at the outermost position
+    std::vector<IterDomain*> non_repeated_loop{
+        ref_tv->getLoopDomain().begin() + 1, ref_tv->getLoopDomain().end()};
```

Comment:
nitpick, should we instead do this instead? vvv
`ref_tv->getLoopDomain().begin() + outermost_pos, ref_tv->getLoopDomain().end()};`

Reply:
I think that should work too, but the only missing ID in the pre-repeat tensors is the outermost ID of the reference loop, so just `+ 1` seems to make more sense to me.